python/ray/autoscaler/autoscaler.py

```diff
@@ -179,11 +183,12 @@ def _update(self):
 
         # Terminate nodes if there are too many
         nodes_to_terminate = []
-        while len(nodes) > self.config["max_workers"]:
+        while (len(nodes) -
+               len(nodes_to_terminate)) > self.config["max_workers"] and nodes:
+            to_terminate = nodes.pop()
             logger.info("StandardAutoscaler: "
-                        "{}: Terminating unneeded node".format(nodes[-1]))
-            nodes_to_terminate.append(nodes[-1])
-            nodes = nodes[:-1]
+                        "{}: Terminating unneeded node".format(to_terminate))
+            nodes_to_terminate.append(to_terminate)
```

Comment:
To clarify, this does the exact same thing right? Please avoid mixing unrelated changes into PRs in the future (or at least comment that this change is just a drive-by refactoring).